v2/pkg/input/uncover.go

```diff
@@ -0,0 +1,152 @@
+package input
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/projectdiscovery/gologger"
+	"github.com/projectdiscovery/ratelimit"
+	ucRunner "github.com/projectdiscovery/uncover/runner"
+	"github.com/projectdiscovery/uncover/uncover"
+	"github.com/projectdiscovery/uncover/uncover/agent/censys"
+	"github.com/projectdiscovery/uncover/uncover/agent/fofa"
+	"github.com/projectdiscovery/uncover/uncover/agent/hunter"
+	"github.com/projectdiscovery/uncover/uncover/agent/quake"
+	"github.com/projectdiscovery/uncover/uncover/agent/shodan"
+	"github.com/projectdiscovery/uncover/uncover/agent/shodanidb"
+	"github.com/projectdiscovery/uncover/uncover/agent/zoomeye"
+)
+
+func GetTargetsFromUncover(delay, limit int, engine, query []string) (chan string, error) {
+
+	uncoverOptions := &ucRunner.Options{
+		Provider: &ucRunner.Provider{},
+		Delay:    delay,
+		Limit:    limit,
+		Query:    query,
+		Engine:   engine,
+	}
+	err := loadProvidersFromEnv(uncoverOptions)
+	if err != nil {
+		return nil, err
+	}
+	var censysRateLimiter, fofaRateLimiter, shodanRateLimiter, shodanIdbRateLimiter, quakeRatelimiter, hunterRatelimiter, zoomeyeRatelimiter *ratelimit.Limiter
+	if uncoverOptions.Delay > 0 {
+		censysRateLimiter = ratelimit.New(context.Background(), 1, time.Duration(uncoverOptions.Delay))
```

Comment:
why are we initializing separate rate-limiters? can't we just use a global one

Reply:
We have a uncover delay flag just for `delay between uncover query requests in seconds.` Removed all the not required ratelimiter. Now using only a single ratelimiter for all the agents.